### DISCUSSION TITLE
Bars shadow

### DISCUSSION THREAD

**Discussion by iDevelopper** [LeoNatan/LNPopupController#485] [General] [answered]
Hello @LeoNatan,

I am encountering some weird things with the shadow of the bars (navigation and tab). As you can see from the images, when the background color of the view table is green and the cell transparent, the navigation bar shadow is blue and the tab bar shadow is yellow. Ditto when I present the popup bar, the shadow of it is yellow.

Do you have an idea ? Thanks.

![image](https://user-images.githubusercontent.com/1782450/138134786-a97300d8-4d15-47f0-bdeb-a7a9ef147dd7.png)

![image](https://user-images.githubusercontent.com/1782450/138134880-f0d78831-7b76-4c58-8167-805f827de924.png)

**Comment by LeoNatan** [comment]
Hello,

I'm not sure what you mean by shadow? For me, the "shadow" is the separator line on the bars. Is this what you mean? I don't see blue or yellow there in the screenshots.

But, you can control the shadow color of each bar from its appearance (including the popup bar).

**Reply by iDevelopper** [reply]
This is what I mean, but according to the documentation I can't achieve to control the shadow from the appearance!

![image](https://user-images.githubusercontent.com/1782450/138140155-05fbca94-94d9-4d74-ba0e-fe45221dfb64.png)

![image](https://user-images.githubusercontent.com/1782450/138140278-5088b978-c6fb-4fdb-b7ee-6f5593feb141.png)

**Reply by LeoNatan** [reply]
What is the code you are trying?

`bar.standardAppearance.shadowColor = .red`
should work.

**Reply by iDevelopper** [reply]
Yes it works for tabBar, not for navigationBar.

And shadowColor = .clear does not work. The bar displays a shadow.

Apple documentation says, about shadowColor:

https://developer.apple.com/documentation/uikit/uibarappearance/3198008-shadowcolor

"When shadowImage is nil, the bar displays a default shadow tinted according to the value of this property. If this property is nil or contains the clear color, the bar displays no shadow."

**Reply by LeoNatan** [reply, marked as answer]
Yes, and it works for me. Navigation bar has a few more appearances. Make sure to update all.
